src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -108,13 +119,35 @@ private void InstallUpdate(UpdatePackage updatePackage)
                                             updateSandboxFolder);
 
                 _logger.Info("Starting update client {0}", _appFolderInfo.GetUpdateClientExePath());
-                _logger.ProgressInfo("NzbDrone will restart shortly.");
+                _logger.ProgressInfo("Sonarr will restart shortly.");
 
                 _processProvider.Start(_appFolderInfo.GetUpdateClientExePath(), GetUpdaterArgs(updateSandboxFolder));
+
+                return;
             }
-            catch (Exception ex)
+            catch (UpdateFailedException ex)
```

Comment:
Previously a generic Exception would be caught by InstallUpdate and logged as Error, and then it swallowed.
Right now I let it get thrown up the stack and caught in the CommandExecutor, which logs it as Error too. So I didn't see any use in catching Exception specifically.

The only difference is that I catch UpdateFailedException and log it as Error, then it gets thrown again, and caught by the Execute(Command command) which turns it into a specific Failure (expected failure) and doesn't actually throw anymore. So all UpdateFailedExceptions get swallowed, but any other are allowed to get thrown all the way up.

I could leave out the entire try catch here and log in ExecuteInstallUpdate instead, if that is preferable.